Neos.Flow/Classes/Security/Authentication/Token/UsernamePassword.php

```diff
@@ -56,6 +78,18 @@ public function updateCredentials(ActionRequest $actionRequest)
         }
     }
 
+    public function getUsername(): string
+    {
```

Comment:
I think this should just be
```php
return $this->credentials['username'] ?? '';
```

Reply:
I lost it here.. how are we suppose to set the values then? If we do not have a place, where we take the request into account, for extractig the username and password?

Reply:
yeah, that kind of misses the point of the interface then to easily be able to overwrite from where of the request the credentials are taken. What if we change the signature to `getUsername(ActionRequest $request): string` as I suggested above?

Reply:
We have the credentials anyways (since we have to satisfy the `TokenInterface` that provides a getter for them).

I'll make a suggestion